src/Uno.UWP/Data/Pdf/PdfPage.Droid.cs

```diff
@@ -0,0 +1,169 @@
+			if (options.DestinationHeight > int.MaxValue)
+				throw new ArgumentOutOfRangeException(nameof(options), $"PdfPageRenderOptions.DestinationHeight = {options.DestinationHeight}. Must be less than or equal to int.MaxValue");
+			#endregion
+
+			return Task.Run(() =>
+			{
+				const int quality = 100;
+				var bitmap = RenderInternal(options);
+				bitmap.Compress(Bitmap.CompressFormat.Png, quality, outputStream.AsStream());
+				return bitmap;
+			}).AsAsyncAction();
+		}
+
+		private Bitmap RenderInternal(PdfPageRenderOptions options)
+		{
+			var destination = new Size(options.DestinationWidth, options.DestinationHeight);
+
+			#region handle 0-width and 0-height
+			if (destination.Width == 0 && destination.Height == 0)
+			{
+				// destination size not set - render with the page original size
+				destination = Size;
+			}
+			else if (destination.Width == 0)
+			{
+				// destination width not set - calculate it based on height proportion
+				double scale = destination.Height / Size.Height;
+				destination = new Size(Size.Width * scale, destination.Height);
+			}
+			else if (destination.Height == 0)
+			{
+				// destination height not set - calculate it based on width proportion
+				double scale = destination.Width / Size.Width;
+				destination = new Size(destination.Width, Size.Height * scale);
+			}
+			#endregion
+
+			#region scale according to DPI
+			float dpi = Graphics.Display.DisplayInformation.GetForCurrentView().LogicalDpi / 96.0f;
+			destination = new Size(destination.Width * dpi, destination.Height * dpi);
+			#endregion
+
+			#region create bitmap
+			var bitmap = Bitmap.CreateBitmap((int)destination.Width, (int)destination.Height, Bitmap.Config.Argb8888);
+			var destinationRect = new Rect(new Foundation.Point(), destination);
+			#endregion
+
+			#region fill with background color
+			using (var canvas = new Canvas(bitmap))
+			{
+				var color = options.BackgroundColor.Equals(default(Color)) ? Colors.White : options.BackgroundColor;
+				canvas.DrawRect(destinationRect, new Paint()
+				{
+					Color = color
+				});
+			}
+			#endregion
+
+			#region render only a portion of the page, defined by PdfPageRenderOptions.SourceRect, if set
+			var sourceRect = options.SourceRect;
+			Matrix matrix = null;
+			if (sourceRect.Width > 0 && sourceRect.Height > 0)
+			{
+				matrix = new Matrix();
+				matrix.SetRectToRect(AsAndroidRectF(sourceRect), AsAndroidRectF(destinationRect), Matrix.ScaleToFit.Start);
+			}
+			#endregion
+
+			// Render content
+			pdfPage.Render(bitmap, null, matrix, PdfRenderMode.ForDisplay);
+			return bitmap;
+		}
+
+		public IAsyncAction PreparePageAsync()
+		{
+			return Task.CompletedTask.AsAsyncAction();
+		}
+
+		public void Dispose()
+		{
+			pdfPage.Close();
+			pdfPage.Dispose();
+		}
+
+		private static RectF AsAndroidRectF(Rect rect) => new RectF(
```

Comment:
Move this here: https://github.com/unoplatform/uno/blob/a6db376e46afd9fb30dee317f08a2e50dc5310e4/src/Uno.Foundation/Rect.Android.cs#L10